readthedocs/projects/tasks.py

```diff
@@ -149,9 +145,15 @@ def run(self, pk, version_pk=None, build_pk=None, record=True,
             self.build_force = force
             self.config = None
 
+            if sync_only:
+                self.sync_repo()
+                return True
```

Comment:
This is something that I talked to @ericholscher, and I saw it very clear at the beginning while we were talking, but the implementation is not as good as I imagined. 

I wanted to avoid the needing of handling the LocalEnvironment inside the `update_imported_docs` and also try to avoid the "jump to a middle step of the flow": `update_imported_docs` is like the third step of the whole `UpdateDocsTask` and was used from outside also -I didn't like that.

Besides, the use of this `sync_only` from outside was done only from https://github.com/rtfd/readthedocs.org/blob/7f6c098983acb4bb3afc90e5c780af7dae9ec9d8/readthedocs/core/views/hooks.py#L106 and that endpoint is deprecated. So, I'd say that we shouldn't be calling this anymore in the future.

----

Regarding your proposals and considering what I wrote on this comment, maybe a Mixin class from where two different Task class inherit is the best approach: we share code, it's cleaner and easy to refactor/remove when finally we don't use the sync task alone anymore.

I will work on this implementation and see how it goes, but it should work :)

Reply:
I did a commit with this change at https://github.com/rtfd/readthedocs.org/pull/3520/commits/c79f9f12fa97436e15fa9b6d357bace3a292aea5

Let me know if it's clearer for you now.

Reply:
Looks good! Another option could be to start using celery's builtin task chaining. Without looking deeply into this I don't know if this would clear up any confusion around the patterns we're using here. For now this accomplishes what we want however, perhaps moving to more common celery patterns can be a later batch of work.